### PR TITLE
Implement 'trustworthy dependency' section

### PR DESCRIPTION
**Pull request by sigurdm**

### CODE REVIEW

File: lib/src/create_report.dart [hunk 1]
```diff
@@ -288,6 +294,122 @@ Future<ReportSection> _followsTemplate(
...
+      }
+    }
+  } on Exception catch (e) {
+    issues.add(_Issue('Could not run pub outdated: $e'));
+    dependenciesUpToDate = false;
+  }
+
+  final currentSdkVersion = Version.parse(Platform.version.split(' ').first);
+  final sdkConstraint =
+      VersionConstraint.parse(pubspec.environment['sdk'] as String);
```
Comment (isoos) [inline lib/src/create_report.dart:339]: Nit: The `_inner` parsed `Pubspec` instance has `VersionConstrained` already.

File: lib/src/create_report.dart [hunk 2]
```diff
@@ -288,6 +294,122 @@ Future<ReportSection> _followsTemplate(
...
+  }
+
+  var allowsCurrentFlutter = true;
+  final flutterVersions = toolEnvironment.runtimeInfo.flutterVersions;
+  if (flutterVersions == null) {
+    issues.add(_Issue(
+        'Found no Flutter in your PATH. Could not determine the current Flutter version.'));
+    allowsCurrentFlutter = false;
+  } else {
+    final usesFlutter = await toolEnvironment.detectFlutterUse(packageDir);
```
Comment (isoos) [inline lib/src/create_report.dart:355]: `pubspec.usesFlutter`?

File: lib/src/create_report.dart [hunk 3]
```diff
@@ -288,6 +294,122 @@ Future<ReportSection> _followsTemplate(
...
+              span: _tryGetSpanFromYamlMap(pubspec.environment, 'flutter'),
+            ),
+          );
+          allowsCurrentFlutter = false;
+        }
+      }
+    }
+  }
+  final publisher = json.decode(await read(
+      'https://pub.dev/api/packages/${pubspec.name}/publisher'))['publisherId'];
```
Comment (isoos) [inline lib/src/create_report.dart:389]: Although it is a very unlikely attack vector on packages already published on pub.dev, let's escape `pubspec.name`.

File: lib/src/create_report.dart (hunk 1 above)
Comment (sigurdm) [reply lib/src/create_report.dart]: Nice, using that instead

File: lib/src/create_report.dart (hunk 2 above)
Comment (sigurdm) [reply lib/src/create_report.dart]: Done

File: lib/src/create_report.dart (hunk 3 above)
Comment (sigurdm) [reply lib/src/create_report.dart]: Done

**Comment by sigurdm** [comment]
We don't have flutter on travis, so my golden files where getting different. I made a hack to disable flutter while generating golden files.

PTAL.

**Pull request merged**